File: app/src/main/java/com/stevenfrew/beatprompter/PlayState.java
package com.stevenfrew.beatprompter;

public enum PlayState {
    AtTitleScreen(0),
    Paused(1),
    Playing(2);

    private int mValue;

    PlayState(int val)
    {
        mValue=val;
    }

    public int asValue()
    {
        return mValue;
    }

    public static PlayState fromValue(int val)
    {
        switch(val)
        {
            case 0:
                return AtTitleScreen;
            case 1:
                return Paused;
            default:
                return Playing;
        }
    }

    static PlayState increase(PlayState state)
    {
        if(state==AtTitleScreen)
            return Paused;
        return Playing;
    }

    static PlayState reduce(PlayState state)
    {
        if(state==Playing)
            return Paused;
        return AtTitleScreen;
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/PlayState.kt
package com.stevenfrew.beatprompter

enum class PlayState constructor(private val mValue: Int) {
    AtTitleScreen(0),
    Paused(1),
    Playing(2);

    fun asValue(): Int {
        return mValue
    }

    companion object {

        fun fromValue(`val`: Int): PlayState {
            return when (`val`) {
                0 -> AtTitleScreen
                1 -> Paused
                else -> Playing
            }
        }

        fun increase(state: PlayState): PlayState {
            return if (state == AtTitleScreen) Paused else Playing
        }

        fun reduce(state: PlayState): PlayState {
            return if (state == Playing) Paused else AtTitleScreen
        }
    }
}